qiskit/primitives/utils.py

```diff
@@ -59,6 +59,14 @@ def init_observable(observable: BaseOperator | str) -> SparsePauliOp:
     if isinstance(observable, SparsePauliOp):
         return observable
     elif isinstance(observable, BaseOperator) and not isinstance(observable, BasePauli):
+        warnings.warn(
```

Comment:
Considere adding a docstring note (similar to https://github.com/Qiskit/qiskit/pull/11520/files#diff-0094a250bbe85d99bff3279e6eaea98adaffcdb59e89773c76cb60be18a19675R157-R161).